client/src/api.ts

```diff
@@ -97,6 +98,94 @@ interface GraphQLResponse<T> {
   data?: T;
 }
 
+export async function fetchMentorByAccessToken(
+  accessToken: string,
+  subject?: string,
+  topic?: string,
+  status?: string
+): Promise<Mentor> {
+  const headers = { Authorization: `bearer ${accessToken}` };
+  const result = await axios.post(
+    GRAPHQL_ENDPOINT,
+    {
+      query: `
+      query {
+        me {
+          mentor {
+            _id
+            name
+            firstName
+            title
+            mentorType
+            lastTrainedAt
+            defaultSubject {
+              _id
+            }
+            subjects {
+              _id
+              name
+              description
+              categories {
+                id
+                name
+                description
+              }
+              topics {
+                id
+                name
+                description
+              }
+              questions {
```

Comment:
pretty sure we don't need most of this either

Reply:
We do if you check carefully